### ISSUE TITLE
Migrate supported Bitcoin testnet network to Testnet4 or Signet

### ISSUE THREAD

**Issue by lumtis** [zeta-chain/node#2242]
**Describe the Issue**

Bitcoin testnet3 might become deprecated and is not optimal for Bitcoin testing (chain much faster than mainnet)

This task is about finding alternative to Bitcoin testnet3 for testnet

We might want to move to Testnet4: https://mempool.space/testnet4

Blocked by https://github.com/zeta-chain/node/issues/2728

We'll be adding Signet support as 1st step to provide a stable test network.

**Comment by ronaldoguedess** [comment]
Please prioritize this: nobody can test with Testnet3 anymore. Here's the link to claim the faucet for Testnet4.

**Claim Faucet for Testnet4**
https://coinfaucet.eu/en/btc-testnet4/

**Comment by fadeev** [comment]
@ronaldoguedess https://github.com/zeta-chain/node/issues/2386

**Comment by ronaldoguedess** [comment]
> We should definitely consider it. It doesn't seem though `btcd` has completed support for testnet4: [btcsuite/btcd#2187](https://github.com/btcsuite/btcd/issues/2187) We might need to consider forking `btcd`

Testnet4 might not be finalized yet, but it's performing 1000% better than Testnet3, which is now deprecated. This makes it the perfect time to add support for Testnet4.

**Comment by jcvernaleo** [comment]
Just so you know, someone is actively working on adding the current testnet4 to btcd now (not just that open issue that we haven't finished).

Hope to have it in within a few days.

**Comment by jcvernaleo** [comment]
I'd of course encourage contributing to btcd rather than forking.  If you tag me in any PRs, I'll make sure to take a look at it sooner rather than later!

**Comment by lumtis** [comment]
@ronaldoguedess we're look into migrating Bitcoin testnet in one of the coming release

Do you have an opinion about using Signet instead of Testnet4?

**Comment by ronaldoguedess** [comment]
I knew signet a few days ago on the mempool, but I believe Testnet4 will require fewer changes and it's newer.

Testnet4 also offers some great faucet tools:

[Mempool Testnet4 Faucet](https://mempool.space/testnet4/faucet)
[CoinFaucet Testnet4](https://coinfaucet.eu/en/btc-testnet4/)

**Comment by ronaldoguedess** [comment]
> @ronaldoguedess we're look into migrating Bitcoin testnet in one of the coming release
> 
> Do you have an opinion about using Signet instead of Testnet4?

I knew signet a few days ago on the mempool, but I believe Testnet4 will require fewer changes and it's newer.

Testnet4 also offers some great faucet tools:

[Mempool Testnet4 Faucet](https://mempool.space/testnet4/faucet)
[CoinFaucet Testnet4](https://coinfaucet.eu/en/btc-testnet4/)

**Comment by ws4charlie** [comment]
Multiple Bitcoin chain zetaclient config will be supported by https://github.com/zeta-chain/node/pull/2870

**Comment by ajtowns** [comment]
Hello; there was a message on #bitcoin-signet irc about wanting some coins, but @CryptoFewka (presumably?) didn't end up hanging around long enough to get an answer:

>     06:12 <CryptoFewka> Hey all, we're looking at integrating Signet in to ZetaChain's
>                         Testnet, and are in need of a stack of sBTC. Ideally we would receive
>                         enough to also disperse it to developers using our SDK to complete
>                         crosschain transactions. Does anyone have any advice on how we could
>                         obtain enough sBTC to enable us to distribute a small amount
>                         ourselves?

FWIW, as I understand it, Babylon set up a discord bot that acted as a faucet for their users during their testing phases, maybe you want to do something similar? Feel free to ping me here or on a new issue here if you want to nail things down without timezones being as much of an issue, maybe?

**Comment by CryptoFewka** [comment]
Hey @ajtowns, my irc client kept dropping in the background and wouldn't reconnect automatically. Tracking the issue here would be great. 

We absolutely would like to operate our own faucet for the tokens, if we could acquire them.

**Comment by ajtowns** [comment]
Sure, point me at an address, and I'll send some "starter capital", then once you've got a faucet setup we can either do a one-off  manual funding, or set it up to receive a distribution from the coinbase funds ([eg](https://mempool.space/signet/tx/acf4fc7f2c69bd799999d93e9ca56b8abc0a256a12adc0251a07a9a9de5214d5)).

**Comment by CryptoFewka** [comment]
Thanks. Can some sBTC please be sent here: `tb1qj06h3t9vh53n960efq477jyywaj2tkyxuk0d3c`?

**Comment by ajtowns** [comment]
https://mempool.space/signet/address/tb1qj06h3t9vh53n960efq477jyywaj2tkyxuk0d3c

**Comment by gartnera** [comment]
Both signet and testnet4 are now live on the ZetaChain athens3 testnet. There's a bit more work to be done on our indexer and in the toolkit but everything has been completed on the node and infrastructure side.
